### PR TITLE
All callbacks now called in a different stack

### LINKED ISSUE #1: Zalgo
Would you be open to a pull request that changed the function invocations to use `process.nextTick` or [asap](https://www.npmjs.com/package/asap) to avoid having callbacks called in the same stack as the previous function?

### PR DESCRIPTION
**Pull request by TehShrike**
Fixes #1

### CODE REVIEW

**Comment by TehShrike** [comment]
Ping <3

File: index.js [hunk 1]
```diff
@@ -5,14 +5,18 @@ var mutexify = function() {
...
     return 0
   }
 
   var release = function(fn, err, value) {
     used = false
     if (queue.length) acquire(queue.shift())
-    if (fn) fn(err, value)
+    if (fn) process.nextTick(function() {
+      fn(err, value)
+    })
```
Comment (mafintosh) [inline index.js]: this one shouldn't be nextTick'ed as it's invoked always sync

Comment (TehShrike) [inline index.js]: That seems wrong to me, but I can change it back.

**Comment by mafintosh** [comment]
Awesome thanks. Out in 1.1.0

**Comment by TehShrike** [comment]
Sweet, thanks!

**Pull request merged**